ui/lib/kv/addon/routes/secret/patch.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import AdapterError from '@ember-data/adapter/error';
+import { breadcrumbsForSecret } from 'kv/utils/kv-breadcrumbs';
+import { service } from '@ember/service';
+export default class SecretPatch extends Route {
+  @service version;
+
+  beforeModel() {
+    if (!this.version.isEnterprise) {
+      throw this.generateError({ message: 'Patching a KV v2 secret is only available on Vault Enterprise.' });
+    }
+  }
+
+  setupController(controller, resolvedModel) {
+    super.setupController(controller, resolvedModel);
+    const breadcrumbsArray = [
+      { label: 'Secrets', route: 'secrets', linkExternal: true },
+      { label: resolvedModel.backend, route: 'list', model: resolvedModel.backend },
+      ...breadcrumbsForSecret(resolvedModel.backend, resolvedModel.path),
+      { label: 'Patch' },
+    ];
+    controller.breadcrumbs = breadcrumbsArray;
+  }
+
+  afterModel(model) {
```

Comment:
Although a user's policy prevents them from accessing links to navigate to patch a secret, this is in case users attempt to navigate directly to the patch route. Without this user would be able to see the page, but their permissions would prevent form submission.
 
<img width="858" alt="Screenshot 2024-08-26 at 5 55 40 PM" src="https://github.com/user-attachments/assets/ed6cd7ae-2f52-4d78-aef6-95f6577e995c">